kubert/src/shutdown/signals.rs

```diff
@@ -0,0 +1,71 @@
+#[cfg(target_os = "linux")]
```

Comment:
To be clear... this isn't Linux specific! This should work on any posix platform. Look to tokio for prior art on properly handling platforms.